src/window/sdl/src/lib.rs

```diff
@@ -28,27 +80,35 @@ use device_gl::Resources;
 /// ```no_run
 /// extern crate gfx_window_sdl;
 /// extern crate sdl2;
-/// 
+///
 /// fn main() {
 ///     let sdl = sdl2::init().unwrap();
-/// 
+///
 ///     let mut builder = sdl.video().unwrap().window("Example", 800, 600);
 ///     let (window, glcontext, device, factory, color_view, depth_view) =
-///         gfx_window_sdl::init(&mut builder);
+///         gfx_window_sdl::init(&mut builder).expect("gfx_window_sdl::init failed!");
 ///
 ///     // some code...
 /// }
 /// ```
-pub fn init(builder: &mut sdl2::video::WindowBuilder) ->
-    (sdl2::video::Window, sdl2::video::GLContext,
-     device_gl::Device, device_gl::Factory,
-     handle::RenderTargetView<Resources, Srgba8>,
-     handle::DepthStencilView<Resources, DepthStencil>)
+pub fn init<Cf>(builder: &mut WindowBuilder) -> Result<InitOk<Cf, DepthStencil>, InitError>
+where
+    Cf: RenderFormat<Channel = Srgb>,
 {
+    // TODO: Support different color channel types and/or other depth formats if possible
     use core::texture::{AaMode, Size};
 
-    let window = builder.opengl().build().unwrap(); //TODO: use actual settings
-    let context = window.gl_create_context().unwrap();
+    let mut window = builder.opengl().build()?;
+
+    let display_mode = DisplayMode {
+        format: sdl2_pixel_format_from_gfx(Cf::get_format())
+                    .ok_or(InitError::PixelFormatUnsupportedError)?,
```

Comment:
Well this was intentional, I indented 4 spaces past the start of the expression for initializing the format field. But I'm perfectly fine with just running rustfmt over everything (even though its rules are probably gonna change soon[-ish]).